subset/bacnet/bacnetTests/src/main/java/PicsTest.java

```diff
@@ -0,0 +1,100 @@
+import com.google.common.collect.Multimap;
+import com.serotonin.bacnet4j.LocalDevice;
+import com.serotonin.bacnet4j.RemoteDevice;
+import com.serotonin.bacnet4j.npdu.ip.IpNetwork;
+import com.serotonin.bacnet4j.service.unconfirmed.WhoIsRequest;
+import helper.*;
+
+import java.util.Map;
+
+public class PicsTest {
+
+  private Connection connection;
+  private BacnetValidation validator;
+  private BacnetPoints bacnetPoints = new BacnetPoints();
+  private String passedTestReport = "RESULT pass protocol.bacnet.pic\n";
+  private String failedTestReport = "RESULT fail protocol.bacnet.pic\n";
+  private String skippedTestReport = "RESULT skip protocol.bacnet.pic\n";
+  private String reportAppendix = "";
+  private String additionalReportAppendix = "";
+  private Csv csv;
+  private static LocalDevice localDevice;
+  private String localIp = "";
+  private String broadcastIp = "";
+  boolean bacnetSupported = false;
+
+  public PicsTest(String localIp, String broadcastIp) throws Exception {
+    this.localIp = localIp;
+    this.broadcastIp = broadcastIp;
+    discoverDevices();
+  }
+
+  private void discoverDevices() throws Exception {
+    connection = new Connection(broadcastIp, IpNetwork.DEFAULT_PORT, localIp);
+    while (!connection.isTerminate()) {
+      localDevice = connection.getLocalDevice();
+      System.err.println("Sending whois...");
```

Comment:
System.out.println ?

Reply:
this is same as in bacnet4j library and Trevor's code